spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/repository/support/DefinitionRepositoryInitializer.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.dataflow.server.repository.support;
+
+import javax.sql.DataSource;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.jdbc.datasource.init.DatabasePopulatorUtils;
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
+import org.springframework.jdbc.support.MetaDataAccessException;
+
+/**
+ * Utility for initializing the Definition Repository's datasource.  If a single
+ * {@link DataSource} is available in the current context, and functionality is enabled
+ * (as it is by default), this will initialize the database.
+ *
+ * By default, initialization of the database can be disabled by configuring the property
+ * <code>spring.cloud.dataflow.server.initialize.enable</code> to false.
+ *
+ * @author Glenn Renfro
+ */
+
+public final class DefinitionRepositoryInitializer implements InitializingBean {
```

Comment:
Is there something within Spring that we can use instead of bringing this file around from project to project (or contribute it to core)?